### DISCUSSION TITLE
Puppeteer + API with params not work - Or one or the other

### DISCUSSION THREAD

**Discussion by rafaelgit** [vercel/community#4981] [Feature Question]
### Question

there is a bug in vercel where puppeteer does not work together with an API that receives parameters.

Or does the API only work
Or does the puppeteer only work

But, if puppeteer is being used in an API that receives parameters, Vercel raises the error as if it did not find chrome. However, this is not true, since puppeteer works without an API that receives parameters:

```
⨯ Error: Could not find Chrome (ver. 119.0.6045.105). This can occur if either
  1. you did not perform an installation before running the script (e.g. `npm install`) or
  2. your cache path is incorrectly configured (which is: /home/sbx_user1051/.cache/puppeteer).
For (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration.
     at ChromeLauncher.resolveExecutablePath (file:///var/task/node_modules/puppeteer-core/lib/esm/puppeteer/node/ProductLauncher.js:260:27)
     at ChromeLauncher.executablePath (file:///var/task/node_modules/puppeteer-core/lib/esm/puppeteer/node/ChromeLauncher.js:197:25)
     at ChromeLauncher.computeLaunchArguments (file:///var/task/node_modules/puppeteer-core/lib/esm/puppeteer/node/ChromeLauncher.js:91:37)
     at async ChromeLauncher.launch (file:///var/task/node_modules/puppeteer-core/lib/esm/puppeteer/node/ProductLauncher.js:53:28)
     at async p (/var/task/.next/server/app/api/generate-pdf/route.js:1:1165)
     at async /var/task/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:41322

```
Step to reproduce:

It works:
```
import { NextResponse } from 'next/server'
import puppeteer from 'puppeteer'

export async function GET() {
   const username = 'Brian'

   const browser = await puppeteer.launch({
     headless: 'new',
     args: ['--no-sandbox', '--disable-setuid-sandbox'],
   })
   const page = await browser.newPage()
   await page.setContent(`<h4>my name is ${username}</h4> `)
   const pdf = await page.pdf({ format: 'A4', printBackground: true })
   await browser.close()

   const headers = new Headers()
   headers.set('Content-Type', 'application/pdf')
   return new NextResponse(pdf, { headers })
}

```
It does not work:
```
import { NextResponse } from 'next/server'
import puppeteer from 'puppeteer'

export async function GET(request: Request) {
   const { searchParams } = new URL(request.url)
   const username = searchParams.get('username')

   const browser = await puppeteer.launch({
     headless: 'new',
     args: ['--no-sandbox', '--disable-setuid-sandbox'],
   })
   const page = await browser.newPage()
   await page.setContent(`<h4>my name is ${username}</h4> `)
   const pdf = await page.pdf({ format: 'A4', printBackground: true })
   await browser.close()

   const headers = new Headers()
   headers.set('Content-Type', 'application/pdf')
   return new NextResponse(pdf, { headers })
}
```

But it's work:
```
import { NextResponse } from 'next/server'

export async function GET(request: Request) {
  const { searchParams } = new URL(request.url)
  const username = searchParams.get('username')

  const pdf = `test ${username}`

  const headers = new Headers()
  headers.set('Content-Type', 'application/pdf')
  return new NextResponse(pdf, { headers })
}

```

in other words, it only gives an error using puppeteer with an API that receives a parameter
I'm using:
- "next": "^14.0.3",
- "typescript": "^5.3.2"
- "puppeteer": "^21.4.1",
